js/weatherApp.js

```diff
@@ -0,0 +1,67 @@
+export const initializeWeatherApp = () => {
+    document.getElementById('weatherCard').addEventListener('dblclick', function() {
+        document.getElementById('weatherModal').showModal();
+        const contentWrapper = document.querySelector('#weatherModal .content-wrapper');
+        contentWrapper.innerHTML = ''; 
+
+        const weatherAppHtml = `
+            <h2>Javascript Weather App</h2>
+            <p>Real-time weather data by location</p>
+            <input type="text" id="cityInput" placeholder="Enter a city">
+            <button id="getWeatherButtonModal">Get Weather</button>
+            <div id="weatherResultModal"></div>
+        `;
+        contentWrapper.insertAdjacentHTML('beforeend', weatherAppHtml);
+
+        window.displayWeatherCard = (weatherData) => {
+            const container = document.getElementById('weatherResultModal');
+            container.innerHTML = '';
+
+            const cardHtml = `
+                <div class="weather-card">
+                    <h2>Weather in ${weatherData.name}</h2>
+                    <p>Temperature: ${kelvinToCelsius(weatherData.main.temp)} °C</p>
+                    <p>Weather: ${weatherData.weather[0].main} - ${weatherData.weather[0].description}</p>
+                    <p>Humidity: ${weatherData.main.humidity}%</p>
+                    <p>Wind Speed: ${weatherData.wind.speed} m/s</p>
+                </div>
+            `;
+
+            container.innerHTML = cardHtml;
+        };
+
+        document.getElementById('getWeatherButtonModal').addEventListener('click', async function() {
+            const city = document.getElementById('cityInput').value;
+            const apiKey = '448f568c5483d5b0f66f39e0a3ac8fbc'; 
```

Comment:
don't expose this on github, create a .env file and import from there, also add that file to .gitignore